conans/client/conf/detect.py

```diff
@@ -147,7 +147,16 @@ def _get_default_compiler(output):
         output.info("CC and CXX: %s, %s " % (cc or "None", cxx or "None"))
         command = cc or cxx
         if "gcc" in command:
-            return _gcc_compiler(output, command)
+            if platform.system() != "Darwin":
+                return _gcc_compiler(output, command)
+
+            # In Mac OS X check if gcc is a fronted using apple-clang
+            _, out = _execute("%s --version" % command)
+            out = out.lower()
+            if "clang" not in out:
+                return _gcc_compiler(output, command)
```

Comment:
The overall logic of this code seems a bit convolute IMHO, with 2 calls to ``_gcc_compiler()`` plus one manual ``_execute("gcc...``. Maybe it would make more sense to put this logic inside ``_gcc_compiler()``, so the compiler is called exactly once, and the rest is processing the output?

Reply:
@memsharded @lasote in that case what about the fallback to clang when gcc is a frontend of ``apple-clang``.

Shall we call ```_clang_compiler(output, "clang")``` inside ```_gcc_compiler(output, command)``` as the logic will be inside ``_gcc_compiler``?